host/linux/aesm.c

```diff
@@ -492,7 +502,7 @@ oe_result_t aesm_get_launch_token(
 
     /* Unpack the response */
     {
-        size_t pos = 0;
+        ssize_t pos = 0;
```

Comment:
Again, why `ssize_t`?

Reply:
Function calls such as in line 224 return `ssize_t`. So instead of casting the return to `size_t`, thought of changing the datatype of `pos` altogether.

Reply:
That's doesn't seem appropriate because ultimately, we don't want `pos` to have negative values (elsewhere, we end up having to cast it back to `size_t` when we do math with it). The only reason the `_unpack_*` functions return any negative value is to indicate failure and it is expected that `pos` is always assigned a non-negative value. We could just as easily denote `#define OE_ERROR_UNPACK ((size_t)-1)` and check for that error state instead of against `-1`. Alternatively as a deeper refactor, we could more consistently implement the `oe_result_t` return pattern instead of returning custom `int`/`ssize_t` from the unpack functions

@mikbras chime in if my interpretation is incorrect.

Reply:
My intention was to use the C++ convention of using (size_t)-1 as the illegal value. I don't like the use of ssize_t.